### PR TITLE
Move PSP to recipe section to improve visibility

### LINKED ISSUE #2817: Include a sample pod security policy
This could be part or #2785 or a standalone sample configuration to illustrate a secure ECK setup when users have turned on the pod security policy admission controller.

### PR DESCRIPTION
**Pull request by pebrc**
Fixes #2817 

This moves the PSP we use in our e2e tests into the recipes section. Adds a README. A few tweaks proposed by @barkbay : restrict to uid >= 1000 for the `fsGroup` and `supplementalGroups` attributes. Still need to verify it works but wanted to share the approach here. There is the caveat that PSPs are already deprecated but given that users ask for it quite regularly it might still make sense to increase the visibility of the PSP we use internally so that it can serve as a starting point for users writing their own. 

README https://github.com/elastic/cloud-on-k8s/blob/c97513225bf8bbd70e59b83264cb5ac0d5c636ab/config/recipes/psp/README.asciidoc

### CODE REVIEW

File: config/recipes/psp/README.asciidoc [hunk 1]
```diff
@@ -0,0 +1,75 @@
+= Secure your cluster with Pod Security Policies
+
+NOTE: PodSecurityPolicy is deprecated in Kubernetes 1.21 and will be removed in 1.25
+
+This recipe contains Pod Security Policies tailored for the use with the Elastic stack. They are to be taken as illustrative examples only and might need adjustment for your specific environment and requirements.  Please read the corresponding link:https://kubernetes.io/docs/concepts/policy/pod-security-policy/[Kubernetes documentation] on how to enable the PodSecurityPolicy admission controller in your cluster.
```
Comment (alaudazzi) [inline config/recipes/psp/README.asciidoc:5]:
```suggestion
This recipe contains Pod Security Policies tailored for the use with the Elastic stack. These are illustrative examples only and might need adjustment for your specific environment and requirements. Read the corresponding link:https://kubernetes.io/docs/concepts/policy/pod-security-policy/[Kubernetes documentation] on how to enable the PodSecurityPolicy admission controller in your cluster.
```

File: config/recipes/psp/README.asciidoc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  - policy
+  resources:
+  - podsecuritypolicies
+  resourceNames:
+  - elastic.restricted
+  verbs:
+  - use
+----
+
+You also have to bind that role to a user or service account or a group of users. Again, the link:https://kubernetes.io/docs/concepts/policy/pod-security-policy/[Kubernetes documentation] has all the details.
```
Comment (alaudazzi) [inline config/recipes/psp/README.asciidoc:33]:
```suggestion
You also have to bind that role to a user or service account or a group of users. See the link:https://kubernetes.io/docs/concepts/policy/pod-security-policy/[Kubernetes documentation] for more details.
```

File: config/recipes/psp/README.asciidoc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  - name: default
+    podTemplate:
+      spec:
+        securityContext:
+          runAsNonRoot: true
+          runAsUser: 1000
+          fsGroup: 1000
+----
+
+Elastic Agent and Beats need wider access to the host and additional Linux capabilities. They therefore need their own Pod Security Policies. Note the different policies below, matching the requirements of the individual Beats:
```
Comment (alaudazzi) [inline config/recipes/psp/README.asciidoc:71]:
```suggestion
Elastic Agent and Beats need wider access to the host and additional Linux capabilities. Therefore, they need their own Pod Security Policies. Note the different policies below, matching the requirements of the individual Beats:
```

**Review by alaudazzi — APPROVED** [review]
LGTM with a few minor edits for the Asciidoc file.

File: config/recipes/psp/README.asciidoc (hunk 2 above)
Comment (barkbay) [inline config/recipes/psp/README.asciidoc:33]:
```suggestion
You also have to bind that role to a user or service account or a group of users. See the link:https://kubernetes.io/docs/concepts/policy/pod-security-policy/#authorizing-policies[Kubernetes documentation] for more details.
```

**Review by barkbay — APPROVED** [review]
LGTM

**Pull request merged**